software/17.gRPC/README.md

```diff
@@ -1,144 +1,264 @@
+### :pushpin: **Tasks**:
+- Create a `main.go` file at the root of your working folder
+- Create a `main` function
+- Check the first argument provided to the program
 
-So here is what you'll have to do:
+  - If it is "-c", it will start the client. For the moment, just print `Starting the client`
+  - If it is "-s", it will start the server. For the moment, just print `Starting the server`
+  - Otherwise, the program will exit with an error code of 1
 
-- In the `src/server.ts` file, you'll find some code. **Don't modify this code**, it is responsible for importing the gRPC package that you'll have to use.
+### :books: **Documentation**:
 
-- Create a `PoChatServer` class and export it as the default export.
-  
-- Its constructor must initialize its `server` member using the `@grpc/grpc-js` package.
+- [Command line args in go](https://gobyexample.com/command-line-arguments)
+
+### :heavy_check_mark: **Validation**:
+Run the following commands:
+```shell
+go run main.go "-c"
+go run main.go "-s"
+```
+Should print respectively:
+```shell
+$ go run main.go "-c"
+Starting the client
+
+
+$ go run main.go "-s"
+Starting the server
+```
+
+And running the following command:
+```shell
+go run main.go "-h" 
+```
+Should print:
+```shell
+$ go run main.go "-h"
+exit status 1
+```
 
-- After the initialization, it must add the pochat.Chat service. For the moment, just print the request data.
+## Step 3 - Implement the server
 
-- Then, you must bind the address `localhost:3000` to the server.
+### :bookmark_tabs: **Description**:
+The protoc compiler generated a `messenger.pb.go` at the first step, which contains a lot of things, including a 
+`MessengerServiceServer` interface.
 
-- Finally, use the `start` method to start the server, still in the constructor.
+During this step, you will implement this interface and its method, `Send`.
 
-> :bulb: The implementation of the service must be cast to a `ChatHandlers` object.
+### :pushpin: **Tasks**:
+- Open the `messenger/messenger.pb.go` file
+- Look for the `MessengerServiceServer`
+- Create a folder named `server`
+- Create a `server.go` in the `server` folder
+- In the `server/server.go` file, create a `Server` structure which contains nothing for the moment
+- Create a `Send` method for the `Server` structure which takes the same arguments as the `Send` method of the 
+  `MessengerServiceServer` structure
+  
+  - Print the content received with the following format: `[SERVER]: received {body} at {date}`
+  - Return the following message to the client: `hello from server !` with the current date as a UNIX timestamp
+
+Now that your `Server` structure implements all the `MessengerServiceServer` methods, you can start the server.
+
+  - Add a `New` method to the `Server` structure
+  - Add a field named `listener` in the `Server` structure
+  - Add a field named `core` in the `Server` structure
```

Comment:
Structure should be called interface if it's go code no ?

Reply:
Nope, it is a structure here :)